src/panels/lovelace/entity-rows/hui-weather-entity-row.ts

```diff
@@ -0,0 +1,261 @@
+
+  @property() private _config?: WeatherRowConfig;
+
+  public setConfig(config: WeatherRowConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = {
+      primary_attribute: "extrema",
+      secondary_attribute: "humidity",
+      ...config,
+    };
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    const pointer =
+      (this._config.tap_action && this._config.tap_action.action !== "none") ||
+      (this._config.entity &&
+        !DOMAINS_HIDE_MORE_INFO.includes(computeDomain(this._config.entity)));
+
+    return html`
+      <state-badge
+        class=${classMap({
+          pointer,
+        })}
+        .hass=${this.hass}
+        .stateObj=${stateObj}
+        .overrideIcon=${weatherIcons[stateObj.state]}
+        @action=${this._handleAction}
+        .actionHandler=${actionHandler({
+          hasHold: hasAction(this._config!.hold_action),
+          hasDoubleClick: hasAction(this._config!.double_tap_action),
+        })}
+        tabindex=${ifDefined(pointer ? "0" : undefined)}
+      ></state-badge>
+      <div
+        class="info ${classMap({
+          pointer,
+        })}"
+        @action=${this._handleAction}
+        .actionHandler=${actionHandler({
+          hasHold: hasAction(this._config!.hold_action),
+          hasDoubleClick: hasAction(this._config!.double_tap_action),
+        })}
+      >
+        <div>
+          ${stateObj.attributes.temperature} ${this._getUnit("temperature")}
+          ${this._config.name || computeStateName(stateObj)}
+        </div>
+        <div class="secondary">
+          ${this.hass.localize(`state.weather.${stateObj.state}`) ||
+            stateObj.state}
+        </div>
+      </div>
+      <div class="info flex-end">
+        <div>
+          ${this._getAttribute(stateObj, this._config.primary_attribute!)}
+        </div>
+        <div>
+          ${this._getAttribute(stateObj, this._config.secondary_attribute!)}
+        </div>
+      </div>
+    `;
+  }
+
+  private _getAttribute(stateObj: any, attribute: string): TemplateResult {
+    const todayForecast = stateObj.attributes.forecast[0];
+
+    if (attribute === "extrema") {
+      return this._getExtrema(todayForecast);
+    }
+
+    const value = todayForecast[attribute]
+      ? todayForecast[attribute]
+      : stateObj.attributes[attribute];
+
+    if (!value) {
```

Comment:
Should I check in the Config if the entity exists? I placed it here because of the default options for the card. Don't want the error if the card was just added